### PR TITLE
Add example on how to test Domain based application on MUnit 2.1

### PR DESCRIPTION
**Pull request by fermujica**
This example was missing in v2.1

### CODE REVIEW

**Review by diegostrubolini — APPROVED** [review]

File: modules/ROOT/pages/domain-based-application-cookbook.adoc [hunk 1]
```diff
@@ -0,0 +1,151 @@
+= Test Domain-Based Applications
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+MUnit 2.x enables you to write MUnit test suites for your Mule applications that share their resources with a Mule domain, without you performing any additional configuration. To understand how this works, it is helpful to consider a specific scenario populated with actual values, and then use those values in steps that illustrate how to write such a test.
```
Comment (dhirsch1) [inline modules/ROOT/pages/domain-based-application-cookbook.adoc]: Remove "you."

File: modules/ROOT/pages/domain-based-application-cookbook.adoc [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+</dependencies>
+----
+
+=====
+
+== Writing an MUnit Test of the Example Domain-Based Application
+
+Because the Mule domain is a dependency of the Mule application, MUnit can start the Mule domain to retrieve the necessary global elements configuration for the application that it's testing.
+
+Assume an MUnit test to validate that the path being pinged is indeed the expected `/api` path:
```
Comment (dhirsch1) [inline modules/ROOT/pages/domain-based-application-cookbook.adoc]: Do we need "is indeed"?

File: modules/ROOT/pages/domain-based-application-cookbook.adoc [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    <munit-tools:assert-equals //<2>
+        actual="#[output text/plain --- payload]" //<3>
+        expected='#["You reached /api"]'/>
+  </munit:validation>
+
+</munit:test>
+----
+
+. Inside the `execution` scope, an HTTP requester sends a `GET` request to your application's endpoint.
+. Inside the `validation` scope, you decide to set up an `assert-equals` processor to assert the response from the `execution` scope against your expected response.
```
Comment (dhirsch1) [inline modules/ROOT/pages/domain-based-application-cookbook.adoc:148]: I don't think "assert" is the right word. Maybe use "compare" instead.

**Review by dhirsch1 — APPROVED** [review]
Approved with a few comments.

File: modules/ROOT/pages/domain-based-application-cookbook.adoc (hunk 3 above)
Comment (fermujica) [reply modules/ROOT/pages/domain-based-application-cookbook.adoc:148]: I'd prefer to leave assert here to keep consistency with the processor's function and name.

**Pull request merged**